pkg/providers/ovirt/validation/validators/network-mapping-validator.go

```diff
@@ -0,0 +1,164 @@
+package validators
+
+import (
+	"fmt"
+
+	"kubevirt.io/client-go/kubecli"
+
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+//NetworkAttachmentDefinitionProvider retrieves NetworkAttachmentDefinition for given name and optional namespace
+type NetworkAttachmentDefinitionProvider interface {
+	Find(name string, namespace string) (*netv1.NetworkAttachmentDefinition, error)
+}
+
+//NetworkMappingValidator provides network mappings validation logic
+type NetworkMappingValidator struct {
+	provider NetworkAttachmentDefinitionProvider
+}
+
+//NewNetworkMappingValidator creates new NetworkMappingValidator that will use provided KubevirtClient
+func NewNetworkMappingValidator(kubevirt kubecli.KubevirtClient) NetworkMappingValidator {
+	return NetworkMappingValidator{
+		provider: &NetworkAttachmentDefinitions{
+			Client: kubevirt.NetworkClient(),
+		},
+	}
+}
+
+//ValidateNetworkMapping validates network mapping
+func (v *NetworkMappingValidator) ValidateNetworkMapping(nics []*ovirtsdk.Nic, mapping *[]v2vv1alpha1.ResourceMappingItem, crNamespace string) []ValidationFailure {
+	var failures []ValidationFailure
+	// Check whether mapping for network is required and was provided
+	if mapping == nil {
+		if v.hasAtLeastOneWithVNicProfile(nics) {
```

Comment:
Well, so we don't care about nic, which has `empty` VNICs, or do we want to create also rules for `empty` networks. Actually I don't really know why `empty` networks was created in oVirt and what's purpose of them, so not sure if it has any value really. So any reason to fail only in case vnicprofile don't exist?

Reply:
Actually, my understanding was that in case described by Moti the nic that is unplugged wouldn't have vnic_profile. Are you saying, that it would have the profile, but the network there wouldn't be present and the check should be `hasAtLeastOneWithNetworkInVNicProfile`?
I wanted to:
 - fail if there is at least one nic that is connected to a network and we have `nil` mapping;
 - allow for configuration with no nics connected to networks and `nil` mapping.

Reply:
empty network / empty vnic profile should not be represented in the target VM.
Simply since the API doesn't support that: The network is mandatory to exist:
https://kubevirt.io/api-reference/v0.26.2/definitions.html#_v1_interface
We should add a warning for this case  - as we discussed originally when reviewed the mapping rules.

Regarding why empty networks was created in oVirt:
The vNIC has two properties: plug and link-state.
plug can be on/off i.e. plugged/unplugged.
link-state can be up/down. So if it is down, the vm remains configured as connected to the network, but temporarily the vm is disconnected from it, without the need to remove the definition of the vnic interface and loose the assigned mac address.